### DISCUSSION TITLE
Compilation error using Python in dbt

### DISCUSSION THREAD

**Discussion by hesteraarts** [dbt-labs/docs.getdbt.com#2028] [Archive]
Hi all,
I just started using the new version of dbt where you can add Python models. I only get a strange compilation error even though I used the exact same code as in the example. Here is my code: 
def model(dbt, session):
    dbt.config(materialized="table")
    activities = dbt.ref("Activities")
return activities.describe()

And here is my error: 
Parsing Error in model marketing (models/staging/Copper/marketing.py)
  In current version, model function should return only one dataframe object

The weird thing is, it worked before, exactly like this. Then I tried to write some Python code in this model, at some point got this error and then deleted all the adjustments. But the error stays. Can someone may be help me out? Thanks in advance!

**Comment by lostmygithubaccount** [comment]
hi @hesteraarts, can you provide more details on:

- data platform (Snowflake, Databricks, BigQuery/GCP)
- version of `dbt` you're running (e.g. `pip list dbt | grep dbt` or `dbt --version`
- the code you're running and the traceback/error? 

the code looks correct. another thing to check is the compiled Python code in something like `target/compiled/{project_name}/models/path/to/model.py`

**Reply by hesteraarts** [reply]
hi! thanks for the reply, I'm using Snowflake and the 1.3 version of dbt. The code I'm running is the code stated in my original message so:
def model(dbt, session):
dbt.config(materialized="table")
activities = dbt.ref("Activities")
return activities.describe()

and below I added the error

![image](https://user-images.githubusercontent.com/104349676/192279927-f7347d6a-cb51-4335-8ceb-cdae1effcbbd.png)


thanks in advance!

**Reply by lostmygithubaccount** [reply]
I'm not able to reproduce this locally or in Cloud. Local:

![image](https://user-images.githubusercontent.com/54814569/192381164-66ccd9ac-88a9-44a4-9038-2ac46f1aa3ce.png)

Cloud:

![image](https://user-images.githubusercontent.com/54814569/192381255-cdb04921-9f76-44c7-8a63-e8d4a108047e.png)

Cloud environment on 1.3:

![image](https://user-images.githubusercontent.com/54814569/192381304-2b0983dd-d1d7-4620-b3e4-9b5b88fe11f5.png)

since you are on Cloud you may want to contact support, but I'm not sure how much help they can offer either. the error message is indicating you're returning more than one dataframe
